creusot-contracts/src/std/thread.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{ghost::local_invariant::Tokens, prelude::*};
+use ::std::thread::JoinHandle;
+
+/// Extension trait for [`JoinHandle`].
+pub trait JoinHandleExt<T> {
+    /// Predicate that specifies the valid return results for the handle.
+    #[logic]
+    fn valid_result(self, x: T) -> bool;
+}
+
+impl<T> JoinHandleExt<T> for JoinHandle<T> {
+    #[logic(opaque)]
+    fn valid_result(self, _x: T) -> bool {
+        dead
+    }
+}
+
+extern_spec! {
+    mod std {
+        mod thread {
+            impl<T> JoinHandle<T> {
+                #[ensures(match result {
+                    Ok(res) => self.valid_result(res),
+                    Err(_) => true,
```

Comment:
That's frustrating: on the one hand, we would like to use `Err(_) => false,` here, because we have proved that the new thread should not panic and because we would like to be able to `unwrap` the result.

On the other hand, this is unsound, because the `Err` case can actually happen in the case of stack overflow or OOM.

I suggest we keep this extern spec as is, and add (via an extension trait) a version that unwraps the result immediately. This will panic if the spawned thread panicked, but this "garbage in, garbage out", so I think this is fine.